client/my-sites/earn/home.tsx

```diff
@@ -504,19 +508,24 @@ export default connect< ConnectedProps, {}, {} >(
 	( state ) => {
 		const selectedSiteSlug = getSelectedSiteSlug( state );
 		const site = getSiteBySlug( state, selectedSiteSlug );
+		const isFreePlan = ! isCurrentPlanPaid( state, site.ID );
+		const hasConnectedAccount = get(
+			state,
+			[ 'memberships', 'settings', site.ID, 'connectedAccountId' ],
+			null
+		);
+		const sitePlanSlug = get( getCurrentPlan( state, site.ID ), 'productSlug', null );
```

Comment:
Also minor but we have an existing selector for this. (note that we normally use selectors in mapStateWithProps, so it's fine to add more if we don't have one already). I'd recommend fetching siteId separately from site as well. It avoids having to deal with a falsey site object if we're still loading data.

```jsx
import { getSelectedSiteId } from 'state/ui/selectors';
import { getSitePlanSlug } from 'state/sites/plans/selectors';
//...

const siteId = getSelectedSiteId( state );
const sitePlanSlug = getSitePlanSlug( state, siteId );
```